app/src/ActionModal/components/ModalVoteSubstrate.vue

```diff
@@ -0,0 +1,428 @@
+        <span>{{ totalVotingPower }}</span>
+        <span>&nbsp;{{ currentNetwork.stakingDenom }}</span>
+      </div>
+    </div>
+    <div class="buttons">
+      <!-- <TmBtn value="Send your Vote" /> -->
+    </div>
+    <TmFormMsg
+      v-if="$v.vote.$error && !$v.vote.required"
+      name="Vote"
+      type="required"
+    />
+  </ActionModal>
+  <!-- </SessionFrame> -->
+</template>
+<script>
+import { mapGetters } from "vuex"
+import { required, decimal } from "vuelidate/lib/validators"
+import { SMALLEST } from "src/scripts/num"
+import ActionModal from "./ActionModal"
+import TmFormGroup from "src/components/common/TmFormGroup"
+import TmField from "src/components/common/TmField"
+// import SessionFrame from "common/SessionFrame"
+import TmBtn from "src/components/common/TmBtn"
+import TmFormMsg from "src/components/common/TmFormMsg"
+import { messageType } from "../../components/transactions/messageTypes"
+import gql from "graphql-tag"
+
+const isValid = (option) => option === `Yes` || option === `No`
+
+export default {
+  name: `modal-vote-substrate`,
+  components: {
+    ActionModal,
+    // SessionFrame,
+    TmFormGroup,
+    TmField,
+    TmFormMsg,
+    TmBtn,
+  },
+  props: {
+    proposalId: {
+      type: [Number, String],
+      required: true,
+    },
+    lastVoteOption: {
+      default: undefined,
+      type: String,
+    },
+  },
+  data: () => ({
+    balances: [],
+    lockedBalance: 0,
+    lockingPeriod: 0,
+    lockingOptions: [
+      { display: `0.1x`, multiplier: 0.1 },
+      { display: `1x`, multiplier: 1 },
+      { display: `2x`, multiplier: 2 },
+      { display: `3x`, multiplier: 3 },
+      { display: `4x`, multiplier: 4 },
+      { display: `5x`, multiplier: 5 },
+      { display: `6x`, multiplier: 6 },
+      { display: `Set Max`, multiplier: 6 },
+    ],
+    selectedLockingOption: { display: `0.1x`, multiplier: 0.1 },
+    totalVotingPower: 0,
+    vote: null,
+    messageType,
+  }),
+  computed: {
+    ...mapGetters([`address`, `currentNetwork`]),
+    transactionData() {
+      return {
+        type: messageType.VOTE,
+        proposalId: this.proposalId,
+        voteOption: this.vote,
+        lockedBalance: Number(this.lockedBalance) || 0,
+        conviction: this.selectedLockingOption
+          ? this.selectedLockingOption.multiplier
+          : 0,
+      }
+    },
+    notifyMessage() {
+      return {
+        title: `Successful vote!`,
+        body: `You have successfully voted ${this.vote} on proposal #${this.proposalId}`,
+      }
+    },
+    stakingDenomBalance() {
+      if (this.balances.length > 0) {
+        return this.balances.find(
+          ({ denom }) => denom === this.currentNetwork.stakingDenom
+        )
+      } else {
+        return {
+          available: 0,
+        }
+      }
+    },
+  },
+  watch: {
+    lockedBalance: {
+      handler() {
+        this.totalVotingController()
+        const input = document.querySelector(".tm-field")
```

Comment:
massive hack. what are you trying to do here?

Reply:
this is to create the cool effect of the TmField/input adapting to the length of what you type. It is really cool

Reply:
@jbibla is there any other way of doing this not using js dom manipulation?

Reply:
oh interesting! 

yeah, i suppose an alternative would be to leave the input 100% width but center the text.

Reply:
yeah, I also thought of this option but the problem there is that how can users then know that there is an input there where they need to write something? :shrug:  leaving the border this is easier to see

Reply:
there will be a flashing vertical line like in all focussed inputs

Reply:
done